src/payment/strategies/adyenv2/adyenv2-payment-strategy.ts

```diff
@@ -262,22 +266,46 @@ export default class AdyenV2PaymentStrategy implements PaymentStrategy {
                             },
                         };
 
-                        onComplete();
+                        if (onComplete) {
+                            onComplete();
+                        }
 
                         resolve(paymentPayload);
                     },
                     onError: (error: AdyenError) => reject(error),
                     size: this._getThreeDS2ChallengeWidgetSize(),
                 });
 
-            const threeDS2Container = this._getAdyenV2PaymentInitializeOptions().threeDS2ContainerId;
+            const { threeDS2ContainerId, additionalActionOptions } = this._getAdyenV2PaymentInitializeOptions();
 
-            onLoad(() => {
-                challengeComponent.unmount();
-                reject();
-            });
+            if (onBeforeLoad) {
+                onBeforeLoad(() => {
```

Comment:
Can you have both `onBeforeLoad` and `onLoad`? But only pass the cancel callback to `onLoad`. i.e.:

```ts
if (onBeforeLoad) {
    onBeforeLoad(AdyenComponentType.IDEAL);
}

challengeComponent.mount(`#${additionalActionOptions.containerId || threeDS2ContainerId}`);

if (onLoad) {
    onLoad(() => {
        challengeComponent.unmount();
        reject();
    }, AdyenComponentType.IDEAL);
}
```